AccessCheckoutSDK/AccessCheckoutSDKTests/client/session/AccessCheckoutClientTests.swift

```diff
@@ -7,10 +7,12 @@ class AccessCheckoutClientTests: XCTestCase {
 
     override func setUp() {
         serviceStubs = ServiceStubs()
+        ServiceDiscoveryProvider.shared.clearCache()
```

Comment:
Could we have a test where the cache is not cleared? This is to make sure that we have a test that could catch a regression if ever we were to change the code the same way it was before where it was crashing due to the .sync() task started in a task started with .async() ?